exasol/bucketfs/__init__.py

```diff
@@ -0,0 +1,148 @@
+    
+        HTTPIE:
+          $  http --auth w:write --auth-type basic GET http://127.0.0.1:6666/default
+
+        CURL:
+          $ curl -i -u "w:write" http://127.0.0.1:6666/default
+"""
+import warnings
+from collections import defaultdict
+from typing import Iterable, Mapping, MutableMapping
+from urllib.parse import urlparse
+
+from exasol_bucketfs_utils_python import BucketFsDeprecationWarning
+from exasol_bucketfs_utils_python.bucket_config import BucketConfig
+from exasol_bucketfs_utils_python.bucketfs_config import BucketFSConfig
+from exasol_bucketfs_utils_python.bucketfs_connection_config import (
+    BucketFSConnectionConfig,
+)
+from exasol_bucketfs_utils_python.buckets import list_buckets
+from exasol_bucketfs_utils_python.list_files import list_files_in_bucketfs
+
+
+class Service:
+    """Provides a simple to use api to access a bucketfs service.
+
+    Attributes:
+        buckets: lists all available buckets.
+    """
+
+    def __init__(self, url: str, credentials: Mapping[str, Mapping[str, str]] = None):
+        """Create a new Service instance.
+
+        Args:
+            url: of the bucketfs service, e.g. `http(s)://127.0.0.1:2580`.
+            credentials: a mapping containing credentials (username and password) for buckets.
+                E.g. {"bucket1": { "username": "foo", "password": "bar" }}
+        """
+        # TODO: Add sanity check for url
+        self._url = url
+        self._authenticator = defaultdict(
+            lambda: {"username": "r", "password": "read"},
+            credentials if credentials is not None else {},
+        )
+
+    @property
+    def buckets(self) -> MutableMapping[str, "Bucket"]:
+        """List all available buckets."""
+        buckets = _list_buckets(self._url)
+        return {
+            name: Bucket(
+                name=name,
+                service=self._url,
+                username=self._authenticator[name]["username"],
+                password=self._authenticator[name]["password"],
+            )
+            for name in buckets
+        }
+
+    def __iter__(self):
+        yield from self.buckets
+
+
+def _list_buckets(
+        url: str,
+) -> Iterable[str]:
+    info = urlparse(url)
+    # suppress warning for users of the new api until the internal migration is done too.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=BucketFsDeprecationWarning)
+        return list_buckets(
+            f"{info.scheme}://{info.hostname}", path=info.path, port=info.port
+        )
+
+
+class Bucket:
+    def __init__(self, name: str, service: str, username: str, password: str):
+        """
+        Create a new bucket instance.
+
+        Args:
+            name: of the bucket.
+            service: url where this bucket is hosted on.
+            username: used for authentication.
+            password: used for authentication.
+        """
+        self._name = name
+        self._service = service
```

Comment:
Sanity check, and I am usually not a fan of non-object parameters, or is this constructor not intended to be used by user? If this is the case, maybe think about to make Bucket an abstract class without an implementation and use an private class for the implementation.